RedisTemplate not present

### Dify version

1.2.0

### dify-spring-boot-starter version

1.8.1

### dify-spring-boot-starter used

dify-spring-boot-starter

### Steps to reproduce

Error creating bean with name 'difyServerToken' defined in class
Caused by: java.lang.TypeNotPresentException: Type org.springframework.data.redis.core.RedisTemplate not present

### ✔️ Expected Behavior

_No response_

### ❌ Actual Behavior

_No response_

--- a/README-zh.md
+++ b/README-zh.md
@@ -41,7 +41,7 @@
         <dependency>
             <groupId>io.github.guoshiqiufeng.dify</groupId>
             <artifactId>dify-bom</artifactId>
-            <version>0.8.2</version>
+            <version>0.8.3</version>
             <type>import</type>
         </dependency>
     </dependencies>

--- a/README.md
+++ b/README.md
@@ -41,7 +41,7 @@ https://guoshiqiufeng.github.io/dify-spring-boot-starter/en
         <dependency>
             <groupId>io.github.guoshiqiufeng.dify</groupId>
             <artifactId>dify-bom</artifactId>
-            <version>0.8.2</version>
+            <version>0.8.3</version>
             <type>import</type>
         </dependency>
     </dependencies>

--- a/dify/spring-boot-starter/dify-spring-boot-starter-autoconfigure/src/main/java/io/github/guoshiqiufeng/dify/autoconfigure/DifyServerAutoConfiguration.java
+++ b/dify/spring-boot-starter/dify-spring-boot-starter-autoconfigure/src/main/java/io/github/guoshiqiufeng/dify/autoconfigure/DifyServerAutoConfiguration.java
@@ -41,17 +41,13 @@ import org.springframework.web.reactive.function.client.WebClient;
  */
 @Slf4j
 @Configuration
-@AutoConfigureAfter(RedisAutoConfiguration.class)
+@AutoConfigureAfter(DifyServerRedisTokenAutoConfiguration.class)
 @ConditionalOnClass({DifyServerClient.class})
 public class DifyServerAutoConfiguration {
 
     @Bean
     @ConditionalOnMissingBean(DifyServerToken.class)
-    public DifyServerToken difyServerToken(ObjectProvider<RedisTemplate<String, String>> redisTemplateProvider) {
-        RedisTemplate<String, String> redisTemplate = redisTemplateProvider.getIfAvailable();
-        if (redisTemplate != null) {
-            return new DifyServerTokenRedis(redisTemplate);
-        }
+    public DifyServerToken difyServerToken() {
         return new DifyServerTokenDefault();
     }
 

--- a/dify/spring-boot-starter/dify-spring-boot-starter-autoconfigure/src/main/java/io/github/guoshiqiufeng/dify/autoconfigure/DifyServerRedisTokenAutoConfiguration.java
+++ b/dify/spring-boot-starter/dify-spring-boot-starter-autoconfigure/src/main/java/io/github/guoshiqiufeng/dify/autoconfigure/DifyServerRedisTokenAutoConfiguration.java
@@ -1,0 +1,37 @@
+package io.github.guoshiqiufeng.dify.autoconfigure;
+
+import io.github.guoshiqiufeng.dify.server.client.DifyServerClient;
+import io.github.guoshiqiufeng.dify.server.client.DifyServerToken;
+import io.github.guoshiqiufeng.dify.server.client.DifyServerTokenDefault;
+import io.github.guoshiqiufeng.dify.server.client.DifyServerTokenRedis;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.data.redis.RedisAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.core.RedisTemplate;
+
+/**
+ * @author yanghq
+ * @version 1.0
+ * @since 2025/4/17 09:30
+ */
+@Configuration
+@ConditionalOnBean(RedisTemplate.class)
+@AutoConfigureAfter(RedisAutoConfiguration.class)
+@ConditionalOnClass({DifyServerClient.class})
+public class DifyServerRedisTokenAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean(DifyServerToken.class)
+    public DifyServerToken difyServerToken(ObjectProvider<RedisTemplate<String, String>> redisTemplateProvider) {
+        RedisTemplate<String, String> redisTemplate = redisTemplateProvider.getIfAvailable();
+        if (redisTemplate != null) {
+            return new DifyServerTokenRedis(redisTemplate);
+        }
+        return new DifyServerTokenDefault();
+    }
+}

--- a/dify/spring-boot-starter/dify-spring-boot-starter-autoconfigure/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports
+++ b/dify/spring-boot-starter/dify-spring-boot-starter-autoconfigure/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports
@@ -1,5 +1,6 @@
 io.github.guoshiqiufeng.dify.autoconfigure.DifyPropertiesAutoConfiguration
 io.github.guoshiqiufeng.dify.autoconfigure.DifyChatAutoConfiguration
 io.github.guoshiqiufeng.dify.autoconfigure.DifyDatasetAutoConfiguration
+io.github.guoshiqiufeng.dify.autoconfigure.DifyServerRedisTokenAutoConfiguration
 io.github.guoshiqiufeng.dify.autoconfigure.DifyServerAutoConfiguration
 io.github.guoshiqiufeng.dify.autoconfigure.DifyWorkflowAutoConfiguration

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,4 +1,4 @@
 # {x-release-please-start-version}
-APP_VERSION=0.8.2
+APP_VERSION=0.8.3
 # {x-release-please-end-version}
 APP_GROUP=io.github.guoshiqiufeng.dify

